### PR TITLE
Add galley metricset for Istio Metricbeat module

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR adds galley metricset for istio Module.

The metrics will be collected from the respective Prometheus exporter endpoint:
`istio-galley.istio-system:15014: The galley job returns the Galley-generated metrics.`

Part of: #15505

### CODE REVIEW

**Comment by ChrsMark** [comment]
cc-ing @sayden, @jsoriano since you reviewed similar ones recently

File: x-pack/metricbeat/module/istio/galley/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  release: beta
+  fields:
+    - name: name
+      type: keyword
+      description: >
+        The name of the resource the metric is related to
+    - name: namespace
+      type: keyword
+      description: >
+        The Kubernates namespace of the resource
```
Comment (jsoriano) [inline x-pack/metricbeat/module/istio/galley/_meta/fields.yml]:
```suggestion
        The Kubernetes namespace of the resource
```

File: x-pack/metricbeat/modules.d/istio.yml.disabled [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+  period: 10s
+  # use istio-galley.istio-system:15014, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
   hosts: ["localhost:15014"]
 
 # Istio pilot. To collect all Pilot-generated metrics.
 - module: istio
   metricsets: ["pilot"]
   period: 10s
-  # use istio-pilot.istio-system:15014, when deploying Metricbeat in a kuberentes cluster
+  # use istio-pilot.istio-system:15014, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/modules.d/istio.yml.disabled]: one more typo before merging 😄

File: x-pack/metricbeat/modules.d/istio.yml.disabled [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
   period: 10s
-  # use istio-telemetry.istio-system:15014, when deploying Metricbeat in a kuberentes cluster
+  # use istio-telemetry.istio-system:15014, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
+  hosts: ["localhost:15014"]
+
+# Istio galley. To collect all Galley-generated metrics
+- module: istio
+  metricsets: ["galley"]
+  period: 10s
+  # use istio-galley.istio-system:15014, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/modules.d/istio.yml.disabled]: kubernetes

File: x-pack/metricbeat/modules.d/istio.yml.disabled [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
-  # use istio-telemetry.istio-system:42422, when deploying Metricbeat in a kuberentes cluster
+  # use istio-telemetry.istio-system:42422, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
   hosts: ["localhost:42422"]
 
 # Istio mixer. To monitor Mixer itself.
 - module: istio
   metricsets: ["mixer"]
   period: 10s
-  # use istio-telemetry.istio-system:15014, when deploying Metricbeat in a kuberentes cluster
+  # use istio-telemetry.istio-system:15014, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/modules.d/istio.yml.disabled]: kubernetes

File: x-pack/metricbeat/modules.d/istio.yml.disabled [hunk 5]
```diff
@@ -1,23 +1,30 @@
 # Module: istio
 # Docs: https://www.elastic.co/guide/en/beats/metricbeat/master/metricbeat-module-istio.html
 
-# Istio mesh. To collect all all Mixer-generated metrics
+# Istio mesh. To collect all Mixer-generated metrics
 - module: istio
   metricsets: ["mesh"]
   period: 10s
-  # use istio-telemetry.istio-system:42422, when deploying Metricbeat in a kuberentes cluster
+  # use istio-telemetry.istio-system:42422, when deploying Metricbeat in a kuberentes cluster as Pod or Daemonset
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/modules.d/istio.yml.disabled]: kubernetes

**Pull request merged**